botorch/acquisition/input_constructors.py

```diff
@@ -634,20 +690,25 @@ def construct_inputs_qNEHVI(
 def get_best_f_analytic(
     training_data: TrainingData,
     objective: Optional[AcquisitionObjective] = None,
```

Comment:
could also raise a warning here and automatically generate the proper `PosteriorTransform` (same below)